rir/src/interpreter/interp.cpp

```diff
@@ -2711,13 +2713,14 @@ SEXP evalRirCode(Code* c, Context* ctx, SEXP* env, const CallContext* callCtxt,
 
 eval_done:
     while (!synthesizeFrames.empty()) {
```

Comment:
I am unclear on this though: when we deopt, we jump here and want to redo everything unoptimized. We get the unoptimized version of the first frame, and evaluate it. Let's say that the second frame is for an inlined function, then by evaluating the unoptimized caller, we actually call the second function. Then we get back here and synthesize the second frame, and evaluate it, for the second time? What is the piece that I am missing?

Reply:
Think about it this way:

Reply:
Original Function:
```
- Inlinee Frame
- Caller Frame
- rest
```

Optimized Function
```
- Combined Frame
- rest
```

On deoptimization we convert it to this:
```
- Callee Frame   # synthesizeFrame = [CallerFrame]
- rest
```

we directly rewrite the top fram, and push every frame but the top frame into the synthesize Frame. Then when the callee returns, instead of returning to `rest` we do this:

```
- Caller Frame  # synthesizeFrame = []
- rest
```

Reply:
Thanks, great explanation. I guess I missed that the evals take a pc, so they don't run from the start.
Now I see only one problem, if the callee frame touches the caller frame, but that we already discussed earlier